src/site/content/en/learn/css/selectors/index.md

```diff
@@ -0,0 +1,486 @@
+
+```css
+article p:first-of-type {
+	color: red;
+	font-size: 1.5em;
+}
+```
+
+CSS provides you with a lot of options to select elements and apply rules to them,
+ranging from very simple to very complex,
+to help solve situations like this.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'XWprGYz'
+} %}
+
+## The parts of a CSS rule
+
+To understand how selectors work and their role in CSS,
+it's important to know the parts of a CSS rule.
+A CSS rule is a block of code,
+containing one or more selectors and one or more declarations.
+
+<figure class="w-figure>
+{% Img src="image/VbAJIREinuYvovrBzzvEyZOpw5w1/Vh2pt6e50XaObt1lEjEv.png", alt="An image of a CSS rule with the selector .my-css-rule", width="800", height="427" %}
+</figure>
+
+In this CSS rule, the **selector** is `.my-css-rule`,
+which finds all elements with a class of `my-css-rule` on the page.
+There are three declarations, within the curly brackets.
+A declaration is a property and value pair which applies styles to the elements matched by the selectors.
+A CSS rule can have as many declarations and selectors as you like.
+
+## Simple selectors
+
+The most straightforward group of selectors target HTML elements plus
+classes, IDs, and other attributes which may be added to an HTML tag.
+
+### Universal selector
+
+A [universal selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Universal_selectors
+)—also known as a wildcard—matches any element.
+
+```css
+* {
+	color: hotpink;
+}
+```
+
+This rule causes every HTML element on the page to have hotpink text.
+
+### Type selector
+
+A [type selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Type_selectors)
+matches a HTML element directly.
+
+```css
+section {
+	padding: 2em;
+}
+```
+
+This rule causes every `<section`> element to have `2em` of `padding` on all sides.
+
+### Class selector
+
+A HTML element can have one or more items defined in their `class` attribute.
+The
+[class selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors)
+matches any element that has that class applied to it.
+
+```html
+<div class="my-class"></div>
+<button class="my-class"></button>
+<p class="my-class"></p>
+```
+
+Any element that has the class applied to it will get colored red:
+
+```css
+.my-class {
+	color: red;
+}
+```
+
+Notice how the `.` is only present in CSS and **not** the HTML.
+This is because the `.` character instructs the CSS language to match class attribute members.
+This is a common pattern in CSS, where a special character,
+or set of characters, is used to define selector types.
+
+A HTML element that has a class of `.my-class` will still be matched to the above CSS rule,
+even if they have several other classes, like this:
+
+```html
+<div class="my-class another-class some-other-class"></div>
+```
+
+This is because CSS looks for a `class` attribute that _contains_ the defined class,
+rather than matching that class exactly.
+
```

Comment:
Maybe add a note that classes and ID's cannot start with a number character?

source: https://www.w3.org/TR/CSS2/syndata.html#:~:text=In%20CSS%2C%20identifiers%20(including%20element,hyphen%20followed%20by%20a%20digit.